control-plane/roles/isolated-clusters/tasks/dns.yaml

```diff
@@ -0,0 +1,27 @@
+- name: Get kubeconfig
+  copy:
+    src: "files/kubeconfigs/kubeconfig.{{ cluster }}"
```

Comment:
About this convention I am not so sure if I am happy. This would at least need to be documented, but then it also motivates long-lived kubeconfigs to be statically put into the deployment, which is probably not so ideal.

As the entire cluster isolation feature requires Gardener anyway, maybe we could do it similar to the way roles like [gardener-monitoring-certs](https://github.com/metal-stack/metal-roles/tree/master/control-plane/roles/gardener-monitoring-certs) do it. So the virtual garden kubeconfig is looked up from to control plane cluster (for which the kubeconfig is usually dynamically created). 

From the virtual garden you can dynamically create an admin kubeconfig for a specific shoot (so `isolated_clusters_partition_services_cluster` needs to be a list of dicts with `shoot_name` and `project_name`). This would work with this filter plugin (the way how to do it is actually taken from gardener docs [here](https://github.com/gardener/gardener/blob/297c6c19ab33751bb27c28ab3d1e41d3706ea918/docs/usage/shoot_access.md#shootsadminkubeconfig-subresource)):

```python
from __future__ import (absolute_import, division, print_function)

__metaclass__ = type

import base64
import yaml
import json

from kubernetes import config, client


def shoot_admin_kubeconfig(kubeconfig, project_namespace, shoot_name, expiry_seconds=28800):
    data = yaml.safe_load(kubeconfig)

    configuration = client.Configuration()
    config.load_kube_config_from_dict(data, client_configuration=configuration)

    api = client.ApiClient(configuration=configuration)

    kubeconfig_request = {
        'apiVersion': 'authentication.gardener.cloud/v1alpha1',
        'kind': 'AdminKubeconfigRequest',
        'spec': {
        'expirationSeconds': expiry_seconds
        }
    }

    response = api.call_api(resource_path=f'/apis/core.gardener.cloud/v1beta1/namespaces/{project_namespace}/shoots/{shoot_name}/adminkubeconfig',
                            method='POST',
                            body=kubeconfig_request,
                            auth_settings=['BearerToken'],
                            _preload_content=False,
                            _return_http_data_only=True,
                        )

    return base64.b64decode(json.loads(response.data)["status"]["kubeconfig"]).decode('utf-8')


class FilterModule(object):
    def filters(self):
        return {
            'shoot_admin_kubeconfig': shoot_admin_kubeconfig,
        }
```

Optionally this could also persist the kubeconfig in a file like `/tmp/kubeconfig.{{ cluster }}`.

What do you think about this?

Reply:
Please don't persist it as a file. It is no longer necessary. You can pass the whole dictionary.